hpy/universal/src/ctx_meth.c

```diff
@@ -52,6 +83,25 @@ ctx_CallRealFunctionFromTrampoline(HPyContext ctx, HPyFunc_Signature sig,
        a->result = f(ctx, _py2h(a->self), h_args, nargs, _py2h(a->kw));
        return;
     }
+    case HPyFunc_GETBUFFERPROC: {
+        HPyFunc_getbufferproc f = (HPyFunc_getbufferproc)func;
+        _HPyFunc_args_GETBUFFERPROC *a = (_HPyFunc_args_GETBUFFERPROC*)args;
+        HPy_buffer hbuf;
+        a->result = f(ctx, _py2h(a->arg0), &hbuf, a->arg2);
+        _buffer_h2py(ctx, &hbuf, a->arg1);
+        HPy_Close(ctx, hbuf.obj);
+        return;
```

Comment:
it's a bit annoying that you need to copy all these fields every time you call the impl. We should remind to check whether it's a performance problem.
A possible hack would be to just cast `Py_Buffer` to `HPy_Buffer` and fix the `obj` before and after the call. This is probably undefined behavior again, but we are already relying on it (see my previous commnet)